src/bpftrace.cpp

```diff
@@ -38,6 +38,7 @@ namespace bpftrace {
 
 DebugLevel bt_debug = DebugLevel::kNone;
 bool bt_verbose = false;
+bool bt_sigint_called = false;
```

Comment:
Could we do this without a global variable? I want to get rid of the other two, but in the meantime not introduce any more. Maybe `bpftrace.request_finalize()` could help?

Reply:
I don't think we can avoid a global here. sigaction's sa_handler is a function pointer (rather than a std::function), which takes a raw function pointer, which can only access global variables (I think they are resolved at compile time). I could probably use a static or a class variable, but any way we do it, I think we would need some global state somewhere (let me know if you prefer the other approaches).

We can't call 'request_finalize' here because that's inside the 'bpftrace' local variable, and we would need a closure to access it (which is not allowed in function pointers).

I would like to avoid globals as well, so if there's a way to do this, please let me know.